File: contests/lc/2273.FindResultantArrayAfterRemovingAnagrams/find_resultant_array_after_removing_anagrams.py
from typing import List

class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        res = []
        prev_key = None
        for w in words:
            key = "".join(sorted(w))
            if key != prev_key:
                res.append(w)
                prev_key = key
        return res

File: contests/lc/2273.FindResultantArrayAfterRemovingAnagrams/find-resultant-array-after-removing-anagrams.go
package _273_FindResultantArrayAfterRemovingAnagrams

import "sort"

func removeAnagrams(words []string) []string {
	result := make([]string, 0, len(words))
	var prevKey string
	hasPrev := false

	for _, w := range words {
		key := sortString(w)
		if !hasPrev || key != prevKey {
			result = append(result, w)
			prevKey = key
			hasPrev = true
		}
	}
	return result
}

func sortString(s string) string {
	r := []rune(s)
	sort.Slice(r, func(i, j int) bool { return r[i] < r[j] })
	return string(r)
}
